### PR TITLE
Cleaned Zest Algorithms

### PR DESCRIPTION
**Pull request by azoitl**
In order to reduce our warning count and increase typed generic use (Issue #155)  this PR typed all generics in the Zest Algorithms package. This reduced the warning count by ap. 200.

### CODE REVIEW

**Comment by azoitl** [comment]
@ptziegler thx for the whitespace clean-up. I rebased this PR on your PR #277 now it is much cleaner.

**Comment by ptziegler** [comment]
> @ptziegler thx for the whitespace clean-up. I rebased this PR on your PR #277 now it is much cleaner.

That was the plan :smile:

File: org.eclipse.zest.layouts/src/org/eclipse/zest/layouts/algorithms/SpringLayoutAlgorithm.java [hunk 1]
```diff
@@ -547,10 +550,10 @@ private void setSprIterationsBasedOnTime() {
 	@Override
 	protected boolean performAnotherNonContinuousIteration() {
 		setSprIterationsBasedOnTime();
-		if (iteration <= sprIterations && largestMovement >= sprMove)
+		if (iteration <= sprIterations && largestMovement >= sprMove) {
 			return true;
-		else
-			return false;
+		}
+		return false;
```
Comment (ptziegler) [inline org.eclipse.zest.layouts/src/org/eclipse/zest/layouts/algorithms/SpringLayoutAlgorithm.java:556]: Wouldn't that be equivalent to:
`return iteration <= sprIterations && largestMovement >= sprMove`

File: org.eclipse.zest.layouts/src/org/eclipse/zest/layouts/algorithms/SpringLayoutAlgorithm.java [hunk 2]
```diff
@@ -790,21 +792,24 @@ private int numRelations(Object src, Object dest) {
...
+		}
+		if (!asynchronous && continueous) {
 			return false;
-		else if (!asynchronous && !continueous)
+		}
+		if (!asynchronous && !continueous) {
 			return true;
+		}
 
 		return false;
```
Comment (ptziegler) [inline org.eclipse.zest.layouts/src/org/eclipse/zest/layouts/algorithms/SpringLayoutAlgorithm.java:814]: That can probably also be simplified to `return asynchronous || !continueous`

File: org.eclipse.zest.layouts/src/org/eclipse/zest/layouts/algorithms/TreeLayoutAlgorithm.java [hunk 3]
```diff
@@ -583,25 +560,28 @@ private int getLevelRecursive(InternalNode layoutEntity, int i, Set seen, Intern
...
 			return true;
-		} else if (!asynchronous && continueous) {
+		}
+		if (!asynchronous && continueous) {
 			return false;
-		} else if (!asynchronous && !continueous) {
+		}
+		if (!asynchronous && !continueous) {
 			return true;
 		}
```
Comment (ptziegler) [inline org.eclipse.zest.layouts/src/org/eclipse/zest/layouts/algorithms/TreeLayoutAlgorithm.java:587]: Same as before, that should be equivalent to `return !continueous;`

File: org.eclipse.zest.layouts/src/org/eclipse/zest/layouts/algorithms/RadialLayoutAlgorithm.java [hunk 4]
```diff
@@ -53,19 +52,23 @@ public RadialLayoutAlgorithm(int styles) {
...
 			return true;
-		else if (!asynchronous && continueous)
+		}
+		if (!asynchronous && continueous) {
 			return false;
-		else if (!asynchronous && !continueous)
+		}
+		if (!asynchronous && !continueous) {
 			return true;
+		}
```
Comment (ptziegler) [inline org.eclipse.zest.layouts/src/org/eclipse/zest/layouts/algorithms/RadialLayoutAlgorithm.java:72]:
```
if (asynchronous && continueous) {
	return false;
}
if (asynchronous && !continueous) {
	return true;
}
if (!asynchronous && continueous) {
	return false;
}
if (!asynchronous && !continueous) {
	return true;
}
return false;
```
=> `return !continueous`

File: org.eclipse.zest.layouts/src/org/eclipse/zest/layouts/algorithms/SpringLayoutAlgorithm.java (hunk 1 above)
Comment (azoitl) [reply org.eclipse.zest.layouts/src/org/eclipse/zest/layouts/algorithms/SpringLayoutAlgorithm.java]: Done

File: org.eclipse.zest.layouts/src/org/eclipse/zest/layouts/algorithms/SpringLayoutAlgorithm.java (hunk 2 above)
Comment (azoitl) [reply org.eclipse.zest.layouts/src/org/eclipse/zest/layouts/algorithms/SpringLayoutAlgorithm.java]: I found this code extra spooky and was to afraid to change it. But you are totally right. Changed it accordingly.

File: org.eclipse.zest.layouts/src/org/eclipse/zest/layouts/algorithms/TreeLayoutAlgorithm.java (hunk 3 above)
Comment (azoitl) [reply org.eclipse.zest.layouts/src/org/eclipse/zest/layouts/algorithms/TreeLayoutAlgorithm.java]: I found this code extra spooky and was to afraid to change it. But you are totally right. Changed it accordingly.

File: org.eclipse.zest.layouts/src/org/eclipse/zest/layouts/algorithms/RadialLayoutAlgorithm.java (hunk 4 above)
Comment (azoitl) [reply org.eclipse.zest.layouts/src/org/eclipse/zest/layouts/algorithms/RadialLayoutAlgorithm.java]: I found this code extra spooky and was to afraid to change it. But you are totally right. Changed it accordingly.

**Review by ptziegler — APPROVED** [review]
Looking good!

**Pull request merged**